src/client/visualizations/bar-chart/improved-bar-chart/interactions/interaction.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2017-2018 Allegro.pl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Datum } from "plywood";
```

Comment:
Looks very generic, is it reusable across different visualisations?

Reply:
Nope, different visualisations have different interactions and different arguments/parameters for specific interactions. I don't think abstracting it would lead to better code.